src/terrama2/services/collector/core/Service.cpp

```diff
@@ -83,7 +83,9 @@ void terrama2::services::collector::core::Service::prepareTask(CollectorId colle
 {
   try
   {
-    taskQueue_.emplace(std::bind(&terrama2::services::collector::core::Service::collect, this, collectorId, *std::dynamic_pointer_cast<CollectorLogger>(logger_), dataManager_));
+    auto collectorLogger = std::dynamic_pointer_cast<CollectorLogger>(logger_->clone());
+    assert(collectorLogger);
```

Comment:
Remove assert and add an exception in case it's not valid
We are using the logger without checking if it's valid in everywhere, so crashes will happen when null